adobe/uix-sdk
# Add version mismatch warnings

## Description
Adds version mismatch detection and logging.

- Modifies the bundler script to embed the version number in `package.json` as a constant.
- Compares `version` strings in the handshake messages used to connect a tunnel.
- Warns (but does not error, for now) on version mismatch

Mismatch logging will happen when initially connecting a document to an extension frame. Both the host and the guest will log version mismatches.



## Related Issue

TODO @fe-lix- may have opened this in JIRA, we've certainly talked about it






## Motivation and Context

Because host apps and guest extensions are almost always two different code bases, they can easily end up using different versions of the SDK. Backwards compatibility may not always be possible, so it's necessary to warn the developer or maintainer when this happens.



## How Has This Been Tested?

Tested by differential builds (building the host and guest SDKs at different versions locally) and then checking the logs in a running example site.

### Testing procedure

Check out this branch and do the following (all commands at repo root)

1. `npm run clean` to remove old build products
2. `npm run demo` to build examples and show current behavior
3. Visit the "Number Discussion" example
4. Open the Console and see _no_ version warnings
5. Ctrl-C to exit demo server
6. `npm pkg set version='99.99.99'` to temporarily set a new version string
7. `npm run build` to build libs with new version string
8. `node scripts/multi-server.mjs production` to run examples _without_ rebuilding them
9. Visit the "Number Discussion" example
10. Open the Console and see version warnings. Examples that have no build process and source the bundles directly will receive the changes, while examples that have build processes will not. This should show detected version mismatches in the console.
11. Ctrl-C to exit demo server
12. `git checkout -- package.json` to revert temporary version change





## Screenshots (if appropriate):
<img width="768" alt="Screenshot 2023-01-04 at 5 21 12 PM" src="https://user-images.githubusercontent.com/1643758/210675061-b8cc0cf9-db09-4aa0-8742-6e1d0f93f831.png">

## Types of changes




## Checklist:
